test/jdk/javax/accessibility/JSlider/AccessibleAction/JSliderAccessibleAction.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8262981
+ * @key headful
+ * @summary Test JSlider Accessibility doAccessibleAction(int)
+ * @run main JSliderAccessibleAction
+ */
+
+import java.awt.AWTException;
+import java.awt.BorderLayout;
+import java.awt.Container;
+import java.awt.GridLayout;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.accessibility.AccessibleAction;
+import javax.accessibility.AccessibleContext;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static java.util.stream.Collectors.toList;
+
+public class JSliderAccessibleAction {
+
+    private static JFrame jFrame;
+    private static JSlider jSlider;
+    private static JButton decrementBtn;
+    private static JButton incrementBtn;
+    private static JButton invalidDecrementBtn;
+    private static JButton invalidIncrementBtn;
+    private static final int INVALID_DECREMENT = 2;
+    private static final int INVALID_INCREMENT = -1;
+    private static final int VALID_DECREMENT = 1;
+    private static final int VALID_INCREMENT = 0;
+    private static volatile AtomicInteger currentJSliderValue;
+    private static volatile AtomicInteger jSliderInitialValue;
+    private static volatile AccessibleAction accessibleAction;
```

Comment:
This field seem to be only accessed from the EDT, similar to UI components above, so why is it volatile and the components are not?

Reply:
Fixed